carbon-design-system/carbon
# fix(combobox): correct behavior of selection onChange with filtered items

Closes https://github.com/carbon-design-system/carbon/issues/16630
Closes https://github.com/carbon-design-system/carbon/issues/16829

This corrects selection and onChange behavior when the listbox items are filtered and one is selected.

#### Changelog

**New**

- Adds `cognateItems` to help test the partial search functionality in Combobox
- Adds tests to cover the defects described in the root issues

**Changed**

- Update allowCustomInput story to log onChange actions to the actions pane
- Update ComboBox to call `filterItems()` instead of using the base `items`

#### Testing / Reviewing

- Go to the Combobox / Allow Custom Input story
- input `apple`
- click to select `pineapple` - selection should be correct, onChange in the actions pane should fire once with the proper value
